### ISSUE TITLE
Rejected Promise does not throw inside Generators if rejected with an undefined error

### ISSUE THREAD

**Issue by luisherranz** [yelouafi/redux-saga#283]
I feel like I must be missing something pretty basic here, but I can't figure out what.

From the docs:
> call(fn, ...args)
>
> Creates an Effect description that instructs the middleware to call the function fn with args as arguments.
> fn: Function - A Generator function, or **normal function which returns a Promise as result**
> args: Array<any> - An array of values to be passed as arguments to fn
>
> fn can be either a normal or a Generator function.
> The middleware invokes the function and examines its result.
> If the result is a Promise, the middleware will suspend the Generator until the Promise is resolved,
> in which case the Generator is resumed with the resolved value. or until the Promise is rejected,
> in which case an error is thrown inside the Generator.

But when I try this:
```javascript
const promiseFn = (flag) => {
    if (flag) return new Promise(resolve => resolve());
    else      return new Promise((r, reject) => reject());
}

function* thisSagaShouldSucceed() {
    try {
        yield call(promiseFn, true);
        console.log('thisSagaShouldSucceed: success!');
    } catch (error) {
        console.log('thisSagaShouldSucceed: fail!');
    }
}

function* thisSagaShouldFail() {
    try {
        yield call(promiseFn, false);
        console.log('thisSagaShouldFail: success!');
    } catch (error) {
        console.log('thisSagaShouldFail: fail!');
    }
}
```

Both generators succeed.

I've tried lots of combinations of `promiseFn` returning promises in different ways, but nothing has worked so far. Or both functions fail, or both functions succeed.

You can play with the real code (and try to fix it if you want) in this tonicdev notebook:
https://tonicdev.com/luisherranz/redux-saga-returning-promise-problem

The same code works fine when using async/await:
https://tonicdev.com/luisherranz/redux-saga-problem-with-async-await

**Comment by luisherranz** [comment]
Ok, I finally got the problem.

This is doesn't throw an error inside the generator:
```javascript
return new Promise((r, reject) => reject());
```

but this does:
```javascript
return new Promise((r, reject) => reject({}));
```

It looks like the async/await version throwing the error even with an argumentless `reject()`:
https://tonicdev.com/luisherranz/redux-saga-problem-with-async-await

So, is this expected? Is this a bug on how `call` is managing promise rejects?

**Comment by yelouafi** [comment]
it's a side effect of the usage of node style callbacks internally

```javascript
cb(err, res) {
  if(!err) // it's an error
  else // it's a success
}
```

since the error is undefined it's interpreted as a success call by the callback

**Comment by luisherranz** [comment]
Thanks @yelouafi.

Redux-saga is genius, by the way. Many thanks for your awesome work.

**Comment by yelouafi** [comment]
@luisherranz Issue was fixed in [0.10.1](https://github.com/yelouafi/redux-saga/releases/tag/v0.10.1)
